Bugsnag/Helpers/BSGDefines.h

```diff
@@ -8,6 +8,28 @@
 #ifndef BSGDefines_h
 #define BSGDefines_h
 
+#include <TargetConditionals.h>
+
 #define BSG_PRIVATE __attribute__((visibility("hidden")))
 
+// Primary capabilities (depends on OS)
+#define BSG_HAVE_APPKIT                       (TARGET_OS_OSX                                                    )
+#define BSG_HAVE_IDENTIFIER_FOR_VENDOR        (                 TARGET_OS_IOS || TARGET_OS_TV                   )
+#define BSG_HAVE_MACH_EXCEPTIONS              (TARGET_OS_OSX || TARGET_OS_IOS                                   )
+#define BSG_HAVE_MACH_THREADS                 (TARGET_OS_OSX || TARGET_OS_IOS || TARGET_OS_TV                   )
+#define BSG_HAVE_OOM_DETECTION                (                 TARGET_OS_IOS || TARGET_OS_TV                   )
```

Comment:
This one is hard to pin down. We currently check for hard crashes, app hangs, and thermal state kills. And if none of those shut it down, we assume OOM. So how many of those possibilities would have to be undeterminable before we say we can't reliably detect OOMs?

Reply:
I would argue it belongs below because this macro does not declare a capability of the OS, it declares whether we enabled the feature. For instance watchOS probably does have an OOM watchdog, but we won't enable our detection because it won't work.

Reply:
The original intent was to keep them separate so that things that depend on previous `BSG_HAVE_XYZ` defines don't end up placed before them when you sort the lines alphabetically.

Reply:
I've changed the comments to be more explicit about why the split exists.